src/components/Publish/Preview/index.tsx

```diff
@@ -4,37 +4,35 @@ import { FormPublishData } from '../_types'
 import { useFormikContext } from 'formik'
 import AssetContent from 'src/components/Asset/AssetContent'
 import { transformPublishFormToDdo } from '../_utils'
-import { Asset } from '@oceanprotocol/lib'
+import { AssetExtended } from 'src/@types/AssetExtended'
 
 export default function Preview(): ReactElement {
-  const [asset, setAsset] = useState<Asset>()
-  const [accessDetails, setAccessDetails] = useState<AccessDetails>()
+  const [asset, setAsset] = useState<AssetExtended>()
   const { values } = useFormikContext<FormPublishData>()
 
   useEffect(() => {
     async function makeDdo() {
-      const asset = await transformPublishFormToDdo(values)
-      setAsset(asset as Asset)
-
+      const asset = (await transformPublishFormToDdo(values)) as AssetExtended
       // dummy BestPrice to trigger certain AssetActions
-      const accessDetails: AccessDetails = {
+      asset.accessDetails = {
```

Comment:
no direct manipulation of React state objects. This should be:


```
const accessDetails = { ... }

setAsset((prevState) => ({
  ...prevState,
  accessDetails
}))
```

Reply:
this doesn't make sense i'm manipulating ` const asset`  and then setting it ` setAsset(asset)`

Reply:
yup you,re right, all good. Was too quick when scanning all changes for direct React state manipulation